Assets/LeapMotion/Scripts/HandPool.cs

```diff
@@ -17,53 +17,94 @@ public class HandPool :
+    public class ModelGroup {
+      public List<IHandModel> modelList;
+      public ModelGroup(List<IHandModel> modelList) {
+        this.modelList = modelList;
+      }
+    }
 
+    private Dictionary<IHandModel, ModelGroup> modelGroupMapping = new  Dictionary<IHandModel, ModelGroup>();
 
     /** Popuates the ModelPool with the contents of the ModelCollection */
     void Start() {
-      ModelPool = new List<IHandModel>();
-      for (int i = 0; i < ModelCollection.Count; i++) {
-        if (ModelCollection[i] != null) {
-          ModelPool.Add(ModelCollection[i]);
-        }
+      ModelPool = new List<ModelGroup>();
+      foreach (ModelPair pair in ModelCollection) {
+        ModelGroup newModelGroup = new ModelGroup(new List<IHandModel>());
+        newModelGroup.modelList.Add(pair.LeftModel);
+        modelGroupMapping.Add(pair.LeftModel, newModelGroup);
+        newModelGroup.modelList.Add(pair.RightModel);
+        modelGroupMapping.Add(pair.RightModel, newModelGroup);
+        ModelPool.Add(newModelGroup);
       }
       controller_ = GetComponent<LeapHandController>();
     }
+    void Update() {
+    }
     /**
      * MakeHandRepresentation receives a Hand and combines that with an IHandModel to create a HandRepresentation
      * @param hand The Leap Hand data to be drive an IHandModel
      * @param modelType Filters for a type of hand model, for example, physics or graphics hands.
      */
     public override HandRepresentation MakeHandRepresentation(Hand hand, ModelType modelType) {
       HandRepresentation handRep = null;
-      for (int i = 0; i < ModelPool.Count; i++) {
-        IHandModel model = ModelPool[i];
-        bool isCorrectHandedness;
-        Chirality handChirality = hand.IsRight ? Chirality.Right : Chirality.Left;
-        isCorrectHandedness = model.Handedness == handChirality;
-        if (!EnforceHandedness || model.Handedness == Chirality.Either) {
-          isCorrectHandedness = true;
-        }
-        bool isCorrectModelType;
-        isCorrectModelType = model.HandModelType == modelType;
-        if (isCorrectModelType && isCorrectHandedness) {
-            ModelPool.RemoveAt(i);
-            handRep = new HandProxy(this, model, hand);
+      List<IHandModel> models = new List<IHandModel>();
+      foreach (ModelGroup group in ModelPool) {
+        for (int i = 0; i < group.modelList.Count; i++) {
+          IHandModel model = group.modelList[i];
+          bool isCorrectHandedness;
+          Chirality handChirality = hand.IsRight ? Chirality.Right : Chirality.Left;
+          isCorrectHandedness = model.Handedness == handChirality;
+          if (!EnforceHandedness || model.Handedness == Chirality.Either) {
+            isCorrectHandedness = true;
+          }
+          bool isCorrectModelType;
+          isCorrectModelType = model.HandModelType == modelType;
+          if (isCorrectModelType && isCorrectHandedness) {
+            group.modelList.RemoveAt(i);
+            --i;
```

Comment:
This line decrementing i does nothing, since we break out of the loop right after.